interfaces/builtin/overmount.go

```diff
@@ -0,0 +1,136 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+// OvermountInterface allows customizing the userspace exposed to the snap
+type OvermountInterface struct{}
+
+func (iface *OvermountInterface) Name() string {
+	return "overmount"
+}
+
+func (iface *OvermountInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+
+	return nil
+}
+
+func (iface *OvermountInterface) SanitizePlug(plug *interfaces.Plug) error {
+	if iface.Name() != plug.Interface {
+		panic(fmt.Sprintf("plug is not of interface %q", iface))
+	}
+
+	// check that we both a source and a destination
+	source, destination, err := iface.getAttribs(plug.Attrs)
```

Comment:
Currently it looks like this:

plugs:
    usrbin:
        interface: overmount
        source: $SNAP_DATA/bin
        destination: /usr/bin

What are you after, in terms of "more generic"?  Are you thinking of this?

plugs:
    usrbin:
        interface: overmount
        mount:
          - source: $SNAP_DATA/bin
            destination: /usr/bin